src/static/js/modules/post-filters-form-validation.js

```diff
@@ -31,34 +36,60 @@ function init() {
     * If year or month is missing, the current month or year are added.
     * @param {string} targetDate Date string to format.
     * @param {string} compDate Other date in the date range we're checking.
+    * @param {Event} event Event object used to prevent form submission on error.
     * @returns {string} Formatted targetDate or an empty string.
     * @throws {Error} If the expected YYYY-MM format is wrong.
     */
-    function formatDateString( targetDate, compDate ) {
+    function formatDateString( targetDate, compDate, event ) {
       var formattedDate = targetDate === '' ? compDate : targetDate;
       if ( formattedDate === '' ) return formattedDate;
 
-      if ( /^\d{4}$/.test( formattedDate ) ) {
-        var month = String( new Date().getMonth() + 1 );
-        if ( month.length !== 2 ) month = '0' + month;
-        formattedDate += '-' + month;
-      } else if ( /^\d{2}$/.test( formattedDate ) ) {
-        var year = new Date().getFullYear();
-        formattedDate = year + '-' + formattedDate;
-      } else if ( /^\d{4}-\d{2}$/.test( formattedDate ) === false ) {
-        var msg = 'Unexpected date format! Should be in YYYY-MM format.';
+      var day = '01';
+      var month = String( new Date().getMonth() + 1 );
+      var year = new Date().getFullYear();
+      var dateArray = ( /(?:(\d{1,2})\/)?(?:(\d{1,2})\/)?(\d{2,4})/ )
+                      .exec( formattedDate );
+      var isArray = Array.isArray || function( obj ) {
+        return toString.call( obj ) === '[object Array]';
+      };
+
+      if ( isArray( dateArray ) ) {
+        day = dateArray[2] || day;
+        month = dateArray[1] || day;
+        year = dateArray[3] || year;
+      } else {
+        event.preventDefault();
+        // TODO: call the notifier when invalid criteria passed.
+        var msg = 'Unexpected date format! Should be in MM/DD/YYYY format.';
         throw new Error( msg );
       }
 
-      return formattedDate;
+      return year + '-' + month + '-' + day;
     }
 
+    /**
+    * @returns {String} Formatted current date string.
```

Comment:
Trivial: `{String}` should be lowercase `{string}` as uppercase designates the function is returning a string created from the String wrapper class, not a string primitive.